lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -976,6 +977,18 @@ SubExprInfo ExprEmitter::visitComb(ConstantOp op) {
   return {Unary, signPreference == RequireSigned ? IsSigned : IsUnsigned};
 }
 
+SubExprInfo ExprEmitter::visitComb(ArraySliceOp op) {
```

Comment:
This needs to make sure the index expression is emitted out of line when non-trivial since you're emitting it twice.  See how `SExtOp` handles this, which has the same issue.